scarlet/frame.py

```diff
@@ -180,99 +185,74 @@ def from_observations(observations, target_psf = None, target_wcs = None, obs_id
-                coord_2obs = coord_frame
-            footprint += (coord_2obs[0] >= -np.finfo(float).eps) * (coord_2obs[1] >= -np.finfo(float).eps) *\
-                (coord_2obs[0] <= obs.frame.shape[-2]) * (coord_2obs[1] <= obs.frame.shape[-1])
-
-        if coverage is 'union':
-            coord_cover = np.where(footprint != 0)
-
-        elif coverage is 'intersection':
-            coord_cover = (np.where(footprint == np.max(footprint)))
-        #Corners of the footprint with padding by the psf size
-        y_fmin = np.min(coord_cover[0]) - fat_pixel_size / 2 + offset[0]
-        x_fmin = np.min(coord_cover[1]) - fat_pixel_size / 2 + offset[1]
-        y_fmax = np.max(coord_cover[0]) + fat_pixel_size / 2 + offset[0]
-        x_fmax = np.max(coord_cover[1]) + fat_pixel_size / 2 + offset[1]
-        # Shape of the resulting frame
-        ny = (y_fmax - y_fmin + 1).astype(int)
-        nx = (x_fmax - x_fmin + 1).astype(int)
-        frame_shape =(len(channels), ny, nx)
-
-        fbox = Box(frame_shape, origin = (0,np.around(y_fmin).astype(int), np.around(x_fmin).astype(int)))
+                if coverage == 'union':
+                    ref_box = ref_box | Box((obs.frame.C, y_max - y_min + 1, x_max - x_min + 1),
+                                            origin = (0, y_min, x_min))
+                elif coverage == 'union':
+                    ref_box = ref_box & Box((obs.frame.C, y_max - y_min + 1, x_max - x_min + 1),
+                                            origin = (0, y_min, x_min))
+
+        _, ny, nx = ref_box.shape
+        frame_shape =(len(channels), np.int((ny + fat_pixel_size)), np.int((nx + fat_pixel_size)))
+        _, o_y, o_x = ref_box.origin
+        fbox = Box(frame_shape, origin = (0, np.int(o_y - fat_pixel_size/2),
+                                          np.int(o_x - fat_pixel_size/2)))
         frame = Frame(fbox, wcs=target_wcs, psfs=target_psf, channels=channels)
-
         # Match observations to this frame
         for obs in observations:
             obs.match(frame)
 
         return frame
 
 
-def get_psf_size(psf):
-    """ Measures the size of a psf by computing the size of the area in 3 sigma around the center.
-
-    This is an approximate method to estimate the size of the psf for setting the size of the frame,
-    which does not require a precise measurement.
-
-    Parameters
-    ----------
-        PSF: `scarlet.PSF` object
-            PSF for whic to compute the size
-    Returns
-    -------
-        sigma3: `float`
-            radius of the area inside 3 sigma around the center in pixels
-    """
-    # Normalisation by maximum
-    psf_frame = psf/np.max(psf)
-
-    # Pixels in the FWHM set to one, others to 0:
-    psf_frame[psf_frame>0.5] = 1
-    psf_frame[psf_frame<=0.5] = 0
-
-    # Area in the FWHM:
-    area = np.sum(psf_frame)
-
-    # Diameter of this area
-    d = 2*(area/np.pi)**0.5
+    def __and__(self, other):
+        """Intersection of two bounding boxes
```

Comment:
Intersection of two `Frame`s?

Reply:
Yes, it's essentially the intersection of the boxes along with handling the channels properly